.github/workflows/ci-unit-tests.yaml

```diff
@@ -0,0 +1,36 @@
+name: Continuous Integration testing for Pull Requests
```

Comment:
From experience, I would recommend a shorter and/or more descriptive name. Maybe "Unit tests"?